CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/parser/SVGParser.java

```diff
@@ -0,0 +1,217 @@
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The SVGParser facilitates the translation of SVGfiles into
+ * an in memory format for easy manipulation and translation. 
+ * In contrast to other Constellation import parsers, this import parser does not 
+ * assume data is in a tabular format and as such will not extend the InportFileParser
+ * 
+ * @author capricornunicorn123
+ */
+public class SVGParser {
+
+    private SVGParser() {
+        throw new IllegalStateException("Utility class");
+    }
+    /**
+     * 
+     * Takes an input stream and translates it to an SVG object. 
+     * Will only translate to an SVG object if the file data is valid.
+     * Does not support multi-line tags.
+     * @param inputStream
+     * @return
+     * @throws IOException 
+     */
+    public static SVGObject parse(InputStream inputStream) throws IOException {
+        SVGObject currentElement = null; 
+        final Collection<SVGObject> roots = new HashSet<>();
+        
+        try (final BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                final String svgElement = SVGParser.isolateSVGElement(line);
+                if (svgElement == null){
+                    continue;
+                }
+                final boolean openTag = SVGParser.isOpenTag(svgElement);
+                final boolean closeTag = SVGParser.isCloseTag(svgElement);
+
+                // Create a new SVGObject with the current SVGObject as the parent 
+                if (openTag) {
+                        
+                    SVGObject newObject = new SVGObject(
+                            SVGParser.getElementType(svgElement), 
+                            currentElement, 
+                            SVGParser.getElementAttributes(svgElement)
+                    );
+                    currentElement = newObject;
+                }
+                // Move back up one level to the curren objects parent
+                if (closeTag) {
+                    currentElement = currentElement.getParent();
+                } 
+                // This parser curently requires all lines with be an SVG tag as it does not support multi line tags.
+                if (!openTag && !closeTag) {
+                    throw new UnsupportedOperationException(String.format("This line could not be interpreted: %s", svgElement));
+                }
+                
+                if (currentElement != null && currentElement.getParent() == null && !roots.contains(currentElement)){
+                    roots.add(currentElement);
+                }
+            }
+        }
+        if (roots.size() != 1) {
+            throw new UnsupportedOperationException(String.format("The SVG file has %s outer elements.", roots.size()));
```

Comment:
not sure exactly what exception to throw here but generally speaking I normally associate `UnsupportedOperationException` being applied to an entire function implementation rather than a portion. Perhaps `IllegalStateException` is better suited here?